examples/example-gauth/pom.xml

```diff
@@ -54,6 +49,11 @@
       <groupId>io.grpc</groupId>
       <artifactId>grpc-auth</artifactId>
     </dependency>
+    <dependency>
```

Comment:
Why need to move this around?

Reply:
Because Maven's dependency resolution is order-dependent; Maven does a breadth-first search and the first time it encounters each dependency decides the version of that dependency it will use. protobuf-java-util pulls in older versions of Guava and error_prone_annotations. Moving it after gRPC happens to fix the resolution order such that gRPC's (direct) dependencies are discovered first. You can see https://source.cloud.google.com/results/invocations/987ebb25-47f0-465d-94d6-4385647b1aef/log for what the error looked like

Reply:
(Note that the only reason we see an error there is because we've enabled the Maven Enforcer plugin with requireUpperBoundDeps; Maven normally will happily downgrade transitive dependencies)